### ISSUE TITLE
E1040 false positive: AWS::Serverless::Function GetAtt Arn rejected for AWS::Lambda::Function.Arn destinations

### ISSUE THREAD

**Issue by jnplthr** [aws-cloudformation/cfn-lint#4649]
### CloudFormation Lint Version

cfn-lint 1.55.1

### What operating system are you using?

Ubuntu (WSL)

### Describe the bug

E1040 fires as a false positive when using !GetAtt <SamFunction>.Arn on an AWS::Serverless::Function resource, to populate any property whose destination format is AWS::Lambda::Function.Arn (e.g. AWS::Config::OrganizationConfigRule's OrganizationCustomRuleMetadata.LambdaFunctionArn).

Related precedent: #4468 / #4469 : same failure shape, fixed for AWS::Lambda::Alias by adding the missing Lambda::Function.Arn format annotation to its schema. Same fix pattern (add "format": "AWS::Lambda::Function.Arn" to the Arn property in the SAM Serverless::Function schema) would likely resolve this.

### Expected behavior

No E1040, since this is the standard documented way to reference a SAM function's ARN.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform: "AWS::Serverless-2016-10-31"
Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: python3.12
      CodeUri: src
  ConfigRule:
    Type: AWS::Config::OrganizationConfigRule
    Properties:
      OrganizationConfigRuleName: example
      OrganizationCustomRuleMetadata:
        LambdaFunctionArn: !GetAtt MyFunction.Arn
        OrganizationConfigRuleTriggerTypes:
          - ScheduledNotification
```

E1040 {'Fn::GetAtt': ['MyFunction', 'Arn']} does not match destination format of 'AWS::Lambda::Function.Arn'

**Comment by prassu017** [comment]
Hi, I'd like to work on this. I reproduced the missing format in the current enhanced-schema artifact: AWS::Serverless::Function.Arn is copied from the raw Lambda schema before extension patches are applied. I plan to fix the SAM generator in resource-provider-enhanced-schemas to copy patched read-only property schemas, add regression coverage, and verify the #4649 template end to end with cfn-lint. Could you assign this to me?

**Comment by jnplthr** [comment]
Hi @prassu017 , I'm not sure if I can assign this to you, probably needs to be done by one of the project owners?

**Comment by kddejong** [comment]
Thanks for the clear report and for pinpointing the fix pattern, @jnplthr — the #4468/#4469 precedent was exactly right.

This is now fixed at the source. `AWS::Serverless::Function`'s `Arn` attribute was missing a `format`, so the `Fn::GetAtt` format check (E1040) treated it as a mismatch against `AWS::Lambda::Function.Arn`. Since SAM transforms `AWS::Serverless::Function` into an `AWS::Lambda::Function`, its `Arn` is a Lambda function ARN, so we added that format to the enhanced schemas:

- aws-cloudformation/resource-provider-enhanced-schemas#39 / #40

Because the fix lives on the attribute's `format`, it covers every destination that expects `AWS::Lambda::Function.Arn` — including your `AWS::Config::OrganizationConfigRule.OrganizationCustomRuleMetadata.LambdaFunctionArn` case, SNS `lambda` subscription endpoints (#4621), and others.

It's in the released schemas now. After `cfn-lint --update-specs`, your reproduction template lints clean on `main` with no rule change — verified end to end.

Closing as fixed. Thanks again to you and @prassu017 for digging into the root cause. 🙏
